### PR TITLE
Feat/884: Capture system_alert when feature-server is online or offline

### PR DESCRIPTION
**Pull request by rammohan-y**
https://github.com/jambonz/jambonz-feature-server/issues/884
Capture system_alert when feature-server is online or offline

### CODE REVIEW

**Comment by rammohan-y** [comment]
@davehorton i had created a draft PR for using writingSystemAlerts when feature server is online/offline. If you can provide your feedback on this, I will incorporate it in other system components like sbc-inbound, sbc-outbound etc.

CC: @vdharashive

File: app.js [hunk 1]
```diff
@@ -25,10 +25,22 @@ const opts = {
...
 const logger = pino(opts, pino.destination({sync: false}));
-const {LifeCycleEvents, FS_UUID_SET_NAME} = require('./lib/utils/constants');
+const {LifeCycleEvents, FS_UUID_SET_NAME, SystemStatusEvents, FEATURE_SERVER} = require('./lib/utils/constants');
 const installSrfLocals = require('./lib/utils/install-srf-locals');
 installSrfLocals(srf, logger);
 
+const writeSystemAlerts = srf.locals?.writeSystemAlerts;
+if (writeSystemAlerts) {
+  writeSystemAlerts({
+    'system_component': FEATURE_SERVER,
```
Comment (davehorton) [inline app.js:35]: please remove the single quotes around the property names

File: app.js [hunk 2]
```diff
@@ -117,13 +129,25 @@ const disconnect = () => {
...
-function handle(signal) {
+async function handle(signal) {
   const {removeFromSet} = srf.locals.dbHelpers;
   srf.locals.disabled = true;
   logger.info(`got signal ${signal}`);
+  const writeSystemAlerts = srf.locals?.writeSystemAlerts;
+  if (writeSystemAlerts) {
+    // it has to be synchronous call, or else by the time system saves the app terminates
+    await writeSystemAlerts({
+      'system_component': FEATURE_SERVER,
```
Comment (davehorton) [inline app.js:143]: ditto above

File: lib/utils/constants.json [hunk 3]
```diff
@@ -222,5 +222,10 @@
   },
   "MAX_SIMRINGS": 10,
   "BONG_TONE": "tone_stream://v=-7;%(100,0,941.0,1477.0);v=-7;>=2;+=.1;%(1400,0,350,440)",
-  "FS_UUID_SET_NAME": "fsUUIDs"
+  "FS_UUID_SET_NAME": "fsUUIDs",
+  "SystemStatusEvents" : {
```
Comment (davehorton) [inline lib/utils/constants.json:226]: I would rather this be named "SystemState".  Also, we need an additional state "GracefulShutdownInProgress".  That needs to be triggered when the cluster scales in -- see utils/aws-sns-lifecycle

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/utils/constants.json (hunk 3 above)
Comment (rammohan-y) [reply lib/utils/constants.json]: @davehorton added GracefulShutdownInProgress event

**Review by davehorton — APPROVED** [review]

**Review by davehorton — APPROVED** [review]

**Comment by davehorton** [comment]
please rebase and fix conflicts

**Comment by rammohan-y** [comment]
> please rebase and fix conflicts

@davehorton rebased

**Pull request merged**